rules/no-unnecessary-polyfills.js

```diff
@@ -0,0 +1,250 @@
+	[MESSAGE_ID_CORE_JS]: 'All polyfilled features imported from `{{coreJsModule}}` are disponible as built-ins. Use the built-ins instead.',
+};
+
+function getTargetsFromPackage(cwd, useBrowserlistFieldByDefault) {
+	const result = readPkgUp.sync({cwd});
+	if (!result || !result.pkg) {
+		return;
+	}
+
+	const {browserlist} = result.pkg;
+	const nodeEngine = result.pkg.engines && result.pkg.engines.node && new SemverNodeVersion(result.pkg.engines.node);
+	return useBrowserlistFieldByDefault ? browserlist || nodeEngine : nodeEngine || browserlist;
+}
+
+const constructorCaseExceptions = {
+	regexp: 'RegExp',
+};
+function constructorCase(name) {
+	return constructorCaseExceptions[name] || upperFirst(camelCase(name));
+}
+
+const additionalPolyfillPatterns = {
+	'es.promise.finally': '|(p-finally)',
+	'es.object.set-prototype-of': '|(setprototypeof)',
+	'es.string.code-point-at': '|(code-point-at)',
+};
+
+const prefixes = '(mdn-polyfills|polyfill-)';
+const suffixes = '(-polyfill)';
+const delimiter = '(\\.|-|\\.prototype\\.|/)?';
+
+const polyfills = Object.keys(compatData).map(feature => {
+	let [ecmaVersion, constructorName, methodName = ''] = feature.split('.');
+
+	if (ecmaVersion === 'es') {
+		ecmaVersion = `(${ecmaVersion}\\d*)`;
+	}
+
+	constructorName = `(${constructorName}|${camelCase(constructorName)})`;
+	if (methodName) {
+		methodName = `(${methodName}|${camelCase(methodName)})`;
+	}
+
+	const methodOrConstructor = methodName || constructorName;
+
+	return {
+		feature,
+		pattern: new RegExp(
+			`^((${prefixes}?`
+				+ `(${ecmaVersion}${delimiter}${constructorName}${delimiter}${methodName}|` // Ex: es6-array-copy-within
+				+ `${constructorName}${delimiter}${methodName}|` // Ex: array-copy-within
+				+ `${ecmaVersion}${delimiter}${constructorName})` // Ex: es6-array
+				+ `${suffixes}?)|`
+				+ `(${prefixes}${methodOrConstructor}|${methodOrConstructor}${suffixes})` // Ex: polyfill-copy-within / polyfill-promise
+				+ `${additionalPolyfillPatterns[feature] || ''})$`,
+			'i',
+		),
+	};
+});
+
+function report(context, node, feature) {
+	let [, namespace, method = ''] = feature.split('.');
+	if (namespace === 'typed-array' && method.endsWith('-array')) {
+		namespace = method;
+		method = '';
+	}
+
+	context.report({
+		node,
+		messageId: MESSAGE_ID_POLYFILL,
+		data: {
+			featureName: `${constructorCase(namespace)}${method && '#'}${camelCase(method)}`,
+		},
+	});
+}
+
+class SemverNodeVersion {
+	constructor(nodeVersion) {
+		this.nodeVersion = nodeVersion;
+		this.validNodeVersion = semver.coerce(nodeVersion);
+	}
+
+	compare(featureVersion) {
+		const supportedNodeVersion = semver.coerce(featureVersion);
+		return this.validNodeVersion
+			? semver.lte(supportedNodeVersion, this.validNodeVersion)
+			: semver.ltr(supportedNodeVersion, this.nodeVersion);
+	}
+}
+
+function processRule(context, node, moduleName, targets) {
+	if (!moduleName || typeof moduleName !== 'string') {
```

Comment:
I think we should only check npm package. So if moduleName starts with `.` should be ignored.